### ISSUE TITLE
T04b(?): Malformed geolocation URI

### ISSUE THREAD

**Issue by jsuvanto** [udacity/ud851-Exercises#268]
https://github.com/udacity/ud851-Exercises/blob/40165eafa14f8c1217c8e8303f2f750d92b65570/app/src/main/java/com/example/android/implicitintents/MainActivity.java#L56

The command at lines 56-59 creates a malformed geolocation URI that doesn't work. Behaviour depends on map software being used, but with Google Maps it shows the last shown location.

You can test this by adding this to line 60:
`Log.d("Address URI", addressUri.toString());`
Debug will show the following line:
`01-13 14:15:27.143 6862-6862/com.example.android.implicitintents D/Address URI: geo:/0%2C0?1600%20Amphitheatre%20Parkway%2C%20CA`

There is an extra slash "/" in the URI, the location "0,0" is encoded (it shouldn't be) and the query string "q=" is missing.

**Comment by ekzor** [comment]
This appears to be because the Udacity course says to use Uri.Builder's `.query` instead of `.appendQueryParameter`. once I changed my code to use the latter method, the map intent started to work properly.

This works:

```
builder.scheme("geo")
    .path("0,0")
    .appendQueryParameter("q", address);
```

Edit: this has apparently already been addressed in an open pull requests #238  and #267

**Comment by johnshea** [comment]
Fixed.
